hanish/zipcode.py

```diff
@@ -0,0 +1,168 @@
+# hanish.zipcode
+# Utility for mapping zip codes to latitude, longitude coords.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Mon May 08 12:11:01 2017 -0400
+#
+# Copyright (C) 2016 Bengfort.com
+# For license information, see LICENSE.txt
+#
+# ID: zipcode.py [] benjamin@bengfort.com $
+
+"""
+Utility for mapping zip codes to latitude, longitude coords. Loads an
+in-memory sqlite database for quickly querying specific zip codes.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import csv
+import sqlite3
+
+from .exceptions import DatabaseError, HanishValueError
+
+
+SCHEMA = (
+    "CREATE TABLE IF NOT EXISTS zipcodes ("
+        "zipcode TEXT, "
+        "latitude REAL, "
+        "longitude REAL"
+    ")"
+)
+
+
+##########################################################################
+## ZipCodeDB
+##########################################################################
+
+class ZipCodeDB(object):
+    """
+    A ZipCodeDB is loaded from a CSV file that maps US Zip Codes to latitutde,
+    longitude coordinates. The data is loaded into an in-memory Sqlite3
+    database for quick querying. The data file can be downloaded from:
+
+        https://gist.github.com/erichurst/7882666
+
+    It is also included in the fixtures directory of the repository.
+
+    Usage:
+
+        # Load the db and lookup coordinates
+        zipdb = ZipCodeDB.load('fixtures/ziplatlon.csv')
+        lat, lon = zipdb.lookup('20001')
+
+        # When done, close the db
+        zipdb.close()
+
+    Note: Zip Codes are strings, not numbers (becasuse of leading zeros).
+    """
+
+    @classmethod
+    def load(klass, path):
```

Comment:
Your suggestion is valid, and worth discussing for this application. I was following more of a `load`/`dump` pattern, with the idea that there was value in an "empty" zipcodes database that could be constructed, then serialized to disk with a `dump` function, and then could be later loaded. So that pattern would be:

```
db = ZipCodeDB() 
# Add stuff to the db 
db.dump(path) 

# Later 
db = ZipCodeDB.load(path) 
```

This is similar to `json.load` and `json.dump` -- but if the database was always tied to the path, then we might as well put the path into the constructor.